random.c

```diff
@@ -421,7 +421,26 @@ random_init(int argc, VALUE *argv, VALUE obj)
 # define USE_DEV_URANDOM 0
 #endif
 
-#if USE_DEV_URANDOM
+#if HAVE_GETENTROPY
+# if defined(__ANDROID__) || defined(__APPLE__)
+#  include <sys/random.h>
+# endif
+# define MAX_SEED_LEN_PER_READ 256
```

Comment:
Why is the reason behind this 256?

Reply:
It s to avoid `EIO` errno case since getentropy can handle 256 bytes maximum per call.

Reply:
The length is described in the man-pages, but no symbol seems defined for it.

macOS:
```
The maximum buffer size permitted is 256 bytes.  If buflen exceeds this, an error of EIO will be indicated.
```
Linux:
```
The maximum permitted value for the length argument is 256.
```